airbyte-webapp/src/hooks/services/Connection/ConnectionFormService.tsx

```diff
@@ -58,17 +66,15 @@ const useConnectionForm = ({ connection, mode, formId, onSubmit, onAfterSubmit,
       setSubmitError(null);
       try {
         // This onSubmit comes from either ReplicationView.tsx (Connection Edit), or CreateConnectionContent.tsx (Connection Create).
-        await onSubmit(formValues);
-
-        formikHelpers.resetForm({ values });
-        // We need to clear the form changes otherwise the dirty form intercept service will prevent navigation
-        clearFormChange(formId);
-
-        onAfterSubmit?.();
-      } catch (e) {
-        if (!(e instanceof ModalCancel)) {
-          setSubmitError(e);
+        const submitResult = await onSubmit(formValues);
+        if ((isSubmitCancel(submitResult) && !submitResult.submitCancel) || !isSubmitCancel(submitResult)) {
+          formikHelpers.resetForm({ values });
+          // We need to clear the form changes otherwise the dirty form intercept service will prevent navigation
+          clearFormChange(formId);
+          onAfterSubmit?.(submitResult);
         }
+      } catch (e) {
+        setSubmitError(e);
```

Comment:
These changes remove the ModalCancel-specific logic as well as pass the result of the submit to the after submit. The latter is specifically so the create connection page knows where to send you.